meson.build

```diff
@@ -1726,6 +1726,14 @@ if conf.get('ENABLE_EFI') == 1 and conf.get('HAVE_BLKID') == 1
                    install_rpath : rootlibexecdir,
                    install : true,
                    install_dir : rootlibexecdir)
+
+        executable('systemd-boot-complete-generator',
```

Comment:
I think this should be 'systemd-boot-update-counter`, which is much more self-explanatory.

Reply:
well, but it actually doesn't "update" it, but actually removes it... It's "sd-boot" that updates it.

Independently of this: the fact that this is a counter appears too much of an implementation detail to me. I mean if the user does "systemctl list-units" and sees "systemd-boot-complete.service", then that is a friendly high-level indication I think, and whether this is a bit or a counter or some kind of other concept shouldn't matter, I think.

Reply:
I still find the name confusing. The generator and the service are named after the time when the service is called, not after it's purpose. Let's consider the service first. It is described as "Mark the Current Boot Loader Entry as Good", so it should be called something like `systemd-mark-boot-entry-as-good.service`, or maybe `systemd-bless-boot-entry.service` (a bit perlesque, but appropriate), or something. Then the generator would be called `systemd-bless-boot-entry-generator` or whatever. The problem I have with current names is that everything is so confusing because everything is named so similarly: `boot-complete.target`, `systemd-boot-complete-generator`, `systemd-boot-complete.service`, even though they serve different purposes and have different levels of generality.